.github/workflows/crawl.yml

```diff
@@ -30,7 +30,9 @@ jobs:
       - uses: ./.github/actions/env
 
       - name: Reattach HEAD (PR)
-        run: git checkout "$(echo ${{ github.head_ref }} | sed -E 's|refs/[a-zA-Z]+/||')"
+        run: git checkout "$(echo "$HEAD_REF" | sed -E 's|refs/[a-zA-Z]+/||')"
+        env:
+          HEAD_REF: "${{ github.head_ref }}"
```

Comment:
this is a safer way of dealing with `head_ref` which is considered a dangerous variable (presumably because it can be controlled by PRs from forks - including here since actionlint was complaining)

Reply:
You can probably also use `GITHUB_HEAD_REF` directly. A lot of context information is also exposed directly as environment variables for scripts.

https://docs.github.com/en/actions/learn-github-actions/variables#default-environment-variables